packages/editor/src/components/properties/EnvMapEditor.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { getDirectoryFromUrl } from '@xrengine/common/src/utils/getDirectoryFromUrl'
+import { useEngineState } from '@xrengine/engine/src/ecs/classes/EngineService'
+import { getComponent } from '@xrengine/engine/src/ecs/functions/ComponentFunctions'
+import { EnvmapComponent } from '@xrengine/engine/src/scene/components/EnvmapComponent'
+import { ErrorComponent } from '@xrengine/engine/src/scene/components/ErrorComponent'
+import { EnvMapSourceType, EnvMapTextureType } from '@xrengine/engine/src/scene/constants/EnvMapEnum'
+import { deserializeEnvMap } from '@xrengine/engine/src/scene/functions/loaders/EnvMapFunctions'
+
+import { setPropertyOnSelectionEntities } from '../../classes/History'
+import ColorInput from '../inputs/ColorInput'
+import CompoundNumericInput from '../inputs/CompoundNumericInput'
+import FolderInput from '../inputs/FolderInput'
+import ImageInput from '../inputs/ImageInput'
+import InputGroup from '../inputs/InputGroup'
+import SelectInput from '../inputs/SelectInput'
+import NodeEditor from './NodeEditor'
+import { EditorComponentType, updateProperty } from './Util'
+
+/**
+ * EnvMapSourceOptions array containing SourceOptions for Envmap
+ */
+const EnvMapSourceOptions = [
+  {
+    label: 'Default',
+    value: EnvMapSourceType.Default
+  },
+  {
+    label: 'Texture',
+    value: EnvMapSourceType.Texture
+  },
+  {
+    label: 'Color',
+    value: EnvMapSourceType.Color
+  },
+  {
+    label: 'None',
+    value: EnvMapSourceType.None
+  }
+]
+
+/**
+ * EnvMapSourceOptions array containing SourceOptions for Envmap
+ */
+const EnvMapTextureOptions = [
+  {
+    label: 'Cubemap',
+    value: EnvMapTextureType.Cubemap
+  },
+  {
+    label: 'Equirectangular',
+    value: EnvMapTextureType.Equirectangular
+  }
+]
+
+/**
+ * EnvMapEditor provides the editor view for environment map property customization.
+ *
+ * @author Nayankumar Patel
+ * @param       props
+ * @constructor
+ */
+export const EnvMapEditor: EditorComponentType = (props) => {
+  const { t } = useTranslation()
+  const entity = props.node.entity
+  const engineState = useEngineState()
+
+  const onChangeCubemapURLSource = useCallback((value) => {
+    const directory = getDirectoryFromUrl(value)
+    if (directory !== envmapComponent.envMapSourceURL) {
+      setPropertyOnSelectionEntities({
+        component: EnvmapComponent,
+        properties: { envMapSourceURL: directory }
+      })
+    }
+  }, [])
+
+  let envmapComponent = getComponent(entity, EnvmapComponent)
+
+  // if component is not there for previously saved model entities then create one
+  if (!envmapComponent) {
+    deserializeEnvMap(props.node.entity, { name: 'envmap', props: { forModel: true } })
```

Comment:
lets use `SCENE_COMPONENT_ENVMAP` instead of `'envmap'` here

Reply:
It is not going to be used but consistency should be first priority. Updated this.